tests/python/gaia-ui-tests/gaiatest/apps/messages/regions/attachment_options.py

```diff
@@ -9,10 +9,9 @@
 
 class AttachmentOptions(Base):
 
-    _attachment_options_locator = (By.CSS_SELECTOR, '#attachment-options[data-type="action"]')
-    _view_button_locator = (By.ID, 'attachment-options-view')
-    _header_locator = (By.CSS_SELECTOR, '.current gaia-header')
-    _cancel_button_locator = (By.ID, 'attachment-options-cancel')
+    _attachment_options_locator = (By.CSS_SELECTOR, 'form.visible[role="dialog"][data-subtype="menu"]')
+    _view_button_locator = (By.CSS_SELECTOR, '[role="dialog"][data-type="action"] > menu > button[data-l10n-id="view-attachment-image"]')
+    _cancel_button_locator = (By.CSS_SELECTOR, '[role="dialog"][data-type="action"] > menu > button:last-child')
```

Comment:
I'd suggest `button[data-l10n-id="cancel"]`. To make sure we're not picking any other 'cancel' button, I would change `tap_cancel()` as:

``` python
    def tap_cancel(self):
        self.root_element.find_element(*self._cancel_button_locator).tap()
        Wait(self.marionette).until(expected.element_not_displayed(self.root_element))

    @property
    def root_element(self):
         return self.marionette.find_element(*self._attachment_options_locator)
```